internal/dynamiccache/cache_test.go

```diff
@@ -238,6 +243,45 @@ func TestCache_Reader(t *testing.T) {
 	})
 }
 
+func TestCache_SampleMetrics(t *testing.T) {
+	c, _, informerMap := setupTestCache(t)
+	c.recorder = metrics.NewRecorder(false)
```

Comment:
Please mock the recorder here and remove the then unused methods
`GetDynamicCacheSizeGvk` and `GetDynamicCacheSizeGvk`